client/keystore/src/lib.rs

```diff
@@ -438,6 +459,29 @@ impl BareCryptoStore for Store {
 	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool {
 		public_keys.iter().all(|(p, t)| self.key_phrase_by_type(&p, *t).is_ok())
 	}
+
+	fn vrf_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &Sr25519Public,
+		label: &'static [u8],
+		prefix: &'static [u8],
+		randomness: &[u8],
+		slot_number: u64,
+		epoch: u64,
```

Comment:
None of `randomness`, `slot_number`, `epoch`, or `threshold` have anything to do with a VRF. They have to do with a use-case of the VRF. This API looks like it is pretending to be general but it is not.